### PR TITLE
docs: optimize contributing guidelines

### PR DESCRIPTION
**Pull request by mariohamann**
## Description:
*PR notes: Please describe the changes in this PR.*

Optimized Contribution Guidelines, merges them with Release Guide and Shows them in Storybook.

## Definition of Reviewable:
*PR notes: Irrelevant elements should be removed.*
- [ ] Documentation is created/updated
- [ ] Migration Guide is created/updated
- [ ] E2E tests (features, a11y, bug fixes) are created/updated
- [ ] Stories (features, a11y) are created/updated
- [ ] relevant tickets are linked
- [ ] PR is assigned to project board

### CODE REVIEW

File: CONTRIBUTING.md
Comment (karlbaumhauer) [reply CONTRIBUTING.md:259]: to prevent multiple updates in the future if we change the process.

File: CONTRIBUTING.md
Comment (abudd1094) [reply CONTRIBUTING.md:65]: It's a little confusing, I agree, could just remove the link.  I think Mario's idea was to lead the users to "explore" components in the Storybook.

**Review by abudd1094 — APPROVED** [review]
Good improvements here.  We can refine further if needed.  I'll give it additional thought before I exit the process based on my own experience.  Thanks for updating!

File: CONTRIBUTING.md
Comment (mariohamann) [reply CONTRIBUTING.md:65]: This was a copy paste thing from the old docs. I'll remove it. 😆

Comment (mariohamann) [inline CONTRIBUTING.md:65]:
```suggestion
  - Provide stories for props, slots etc. in Storybook
```

File: CONTRIBUTING.md
Comment (mariohamann) [reply CONTRIBUTING.md]: I improved it. Could you have a look again? If it doesn't fit, feel free to rewrite. (It was just copy paste. :D)

File: CONTRIBUTING.md
Comment (karlbaumhauer) [inline CONTRIBUTING.md:65]: would it make sense (as the point has been mentioned in #622) to explain a bit more detailed what we are using the stories for?

File: CONTRIBUTING.md
Comment (karlbaumhauer) [inline CONTRIBUTING.md:232]:
```suggestion
> - `docs`: This type is set when the last commit message starts with `docs:` and deploys only the Storybook to the CDN
```

File: CONTRIBUTING.md
Comment (karlbaumhauer) [inline CONTRIBUTING.md:231]:
```suggestion
> - `code`: This type is set when the last commit message starts with `chore(release/components):` (coming from a semantic-release version bump) and it deploys everything to the CDN
```

File: CONTRIBUTING.md
Comment (karlbaumhauer) [reply CONTRIBUTING.md]: added 2 suggestions but rather than this, its fine. thx.

**Review by karlbaumhauer — CHANGES_REQUESTED** [review]

**Comment by abudd1094** [comment]
@karlbaumhauer to continue the conversation from Azra's ticket...

To fix the "confusion" we would need an even better CONTRIBUTING.md and more attentive reviews.  Frankly I don't find this structure effective to communicate what I now understand about this project.

We have a combination of technologies (Shoelace, Storybook, Chromatic, etc...) and it is not self-explanatory how they come together.  Perhaps it was clear at the conception among the members of the core team but not to any others who join. 

The most effective set of guidelines would reduce wordiness around explaining concepts and communicate concisely each consideration of the developer.  This includes prioritzing the explanation of technologies and how they are expected to be used.  For the topic of stories, an explicit section in CONTRIBUTING.md:

## Stories
We write stories primarily as visual tests for [Chromatic](https://www.chromatic.com/).  The idea is to create screenshots of every possible visual scenario so we can easily review when there may be breaking changes.  Additionally, we have a limited number of "screenshots" in Chromatic, so it is best to minimize the total amount of stories.  Do this by combining different visual tests with the use of our "Storybook Helpers" axis properties.  (Provide some markup & examples).

This short paragraph would have made it 100% clear to me from day one what the stories are supposed to be about.  Make it easy, section title "Stories", immediately explain the purpose in this context.  

I'll leave the project by December 18th latest, and need to reduce a few days in between based on my current utilization.  I could elaborate on my own ideal guidelines based on my experience here if you think that would be helpful.

**Comment by karlbaumhauer** [comment]
> @karlbaumhauer to continue the conversation from Azra's ticket...
> 
> To fix the "confusion" we would need an even better CONTRIBUTING.md and more attentive reviews. Frankly I don't find this structure effective to communicate what I now understand about this project.
> 
> We have a combination of technologies (Shoelace, Storybook, Chromatic, etc...) and it is not self-explanatory how they come together. Perhaps it was clear at the conception among the members of the core team but not to outsiders.
> 
> The most effective set of guidelines would reduce wordiness around explaining concepts and communicate concisely each consideration of the developer. This includes prioritzing the explanation of technologies and how they are expected to be used. For the topic of stories, an explicit section in CONTRIBUTING.md:
> 
> ## Stories
> We write stories primarily as visual tests for [Chromatic](https://www.chromatic.com/). The idea is to create screenshots of every possible visual scenario so we can easily review when there may be breaking changes. Additionally, we have a limited number of "screenshots" in Chromatic, so it is best to minimize the total amount of stories. Do this by combining different visual tests with the use of our "Storybook Helpers" axis properties. (Provide some markup & examples).
> 
> This short paragraph would have made it 100% clear to me from day one what the stories are supposed to be about. Make it easy, section title "Stories", immediately explain the purpose in this context.
> 
> Currently we have a mix of language based more around concepts rather than explicit development guidelines: image
> 
> I think Mario did well here elaborating on the existing guidelines, but it needs more for outsiders to gain quick and effective understanding. I'll leave the project by December 18th latest, and need to reduce a few days in between based on my current utilization. I could draft my own ideal guidelines based on my experience here if you think that would be helpful. I would approve this and create a new ticket if so.

thx @abudd1094 for the explanation. My suggestion: As soon as @mariohamann merged this PR, you branch off `main` and push your suggestion. Afterwards, you & me are going through it and in case you can finish it within your time here in this project, its fine otherwise I will take care of it and might just ask you for short feedback afterwards.

**Review by yoezlem — APPROVED** [review]

**Comment by mariohamann** [comment]
@karlbaumhauer I integrated all suggestions and allowed me to merge. :)

**Pull request merged**
